src/vos/vos_dtx.c

```diff
@@ -3249,3 +3249,47 @@ vos_dtx_renew_epoch(struct dtx_handle *dth)
 	if (dth->dth_local_stub != NULL)
 		lrua_refresh_key(dth->dth_local_stub, dth->dth_epoch);
 }
+
+int
+vos_dtx_local_begin(struct dtx_handle *dth, daos_handle_t poh)
+{
+	struct vos_pool      *pool;
+	struct umem_instance *umm;
+	int                   rc;
+
+	if (dth == NULL || !dth->dth_local) {
+		return 0;
+	}
+
+	/**
+	 * RDB is intended as the main user of local transactions. RDB is known
+	 * to engage over 32 OIDs per transaction. Hence, the initial value is
+	 * hoped to accommodate all of them.
+	 */
+	dth->dth_local_oid_cap = (1 << 6);
+	dth->dth_local_oid_cnt = 0;
+	D_ALLOC_ARRAY(dth->dth_local_oid_array, dth->dth_local_oid_cap);
```

Comment:
In other places, to avoid having allocation on the critical path, we would have some embedded ones and only do allocation if we go over the embedded amount

Reply:
@liw hinted that RDB uses somewhere between 32 and 64 OIDs per transaction. I assumed having a 64-slot embedded array is a little bit too much so I went with having a 64-slot allocated default array which I believe should allow RDB to go with a single alloc per transaction. No reallocs.